File: example/java_ls/config/config/task/Task2.java
package config.task;

public class Task2 {
    private int taskid;
    private java.util.List<String> name;
    private int nexttask;
    private config.task.Completecondition completecondition;
    private int exp;
    private boolean testBool;
    private String testString;
    private config.Position testStruct;
    private java.util.List<Integer> testList;
    private java.util.List<config.Position> testListStruct;
    private java.util.List<config.ai.TriggerTick> testListInterface;
    private config.task.Taskextraexp NullableRefTaskid;
    private config.task.Task NullableRefNexttask;

    private Task2() {
    }

    public static Task2 _create(configgen.genjava.ConfigInput input) {
        Task2 self = new Task2();
        self.taskid = input.readInt();
        {
            int c = input.readInt();
            if (c == 0) {
                self.name = java.util.Collections.emptyList();
            } else {
                self.name = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.name.add(input.readTextInPool());
                }
            }
        }
        self.nexttask = input.readInt();
        self.completecondition = config.task.Completecondition._create(input);
        self.exp = input.readInt();
        self.testBool = input.readBool();
        self.testString = input.readStringInPool();
        self.testStruct = config.Position._create(input);
        {
            int c = input.readInt();
            if (c == 0) {
                self.testList = java.util.Collections.emptyList();
            } else {
                self.testList = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.testList.add(input.readInt());
                }
            }
        }
        {
            int c = input.readInt();
            if (c == 0) {
                self.testListStruct = java.util.Collections.emptyList();
            } else {
                self.testListStruct = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.testListStruct.add(config.Position._create(input));
                }
            }
        }
        {
            int c = input.readInt();
            if (c == 0) {
                self.testListInterface = java.util.Collections.emptyList();
            } else {
                self.testListInterface = new java.util.ArrayList<>(c);
                for (; c > 0; c--) {
                    self.testListInterface.add(config.ai.TriggerTick._create(input));
                }
            }
        }
        return self;
    }

    /**
     * 任务完成条件类型（id的范围为1-100）
     */
    public int getTaskid() {
        return taskid;
    }

    public java.util.List<String> getName() {
        return name;
    }

    public int getNexttask() {
        return nexttask;
    }

    public config.task.Completecondition getCompletecondition() {
        return completecondition;
    }

    public int getExp() {
        return exp;
    }

    public boolean getTestBool() {
        return testBool;
    }

    public String getTestString() {
        return testString;
    }

    public config.Position getTestStruct() {
        return testStruct;
    }

    public java.util.List<Integer> getTestList() {
        return testList;
    }

    public java.util.List<config.Position> getTestListStruct() {
        return testListStruct;
    }

    public java.util.List<config.ai.TriggerTick> getTestListInterface() {
        return testListInterface;
    }

    public config.task.Taskextraexp nullableRefTaskid() {
        return NullableRefTaskid;
    }

    public config.task.Task nullableRefNexttask() {
        return NullableRefNexttask;
    }

    @Override
    public String toString() {
        return "(" + taskid + "," + name + "," + nexttask + "," + completecondition + "," + exp + "," + testBool + "," + testString + "," + testStruct + "," + testList + "," + testListStruct + "," + testListInterface + ")";
    }

    public void _resolveDirect(config.ConfigMgr mgr) {
        NullableRefTaskid = mgr.task_taskextraexp_All.get(taskid);
        NullableRefNexttask = mgr.task_task_All.get(nexttask);
    }

    public void _resolve(config.ConfigMgr mgr) {
        completecondition._resolve(mgr);
        _resolveDirect(mgr);
    }

    public static Task2 get(int taskid) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getTaskTask2(taskid);
    }

    public static java.util.Collection<Task2> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allTaskTask2();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Task2 self = Task2._create(input);
                mgr.task_task2_All.put(self.taskid, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            for (Task2 e : mgr.task_task2_All.values()) {
                e._resolve(mgr);
            }
        }

    }

}

File: example/lua_ls_client/cfg/task/task2.lua
local cfg = require "cfg._cfgs"
local Beans = cfg._beans

---@class cfg.task.task2
---@field taskid number , 任务完成条件类型（id的范围为1-100）
---@field name table<number,text> 
---@field nexttask number 
---@field completecondition Beans.task.completecondition 
---@field exp number 
---@field testBool boolean 
---@field testString string 
---@field testStruct Beans.position 
---@field testList table<number,number> 
---@field testListStruct table<number,Beans.position> 
---@field testListInterface table<number,Beans.ai.triggertick> 
---@field get fun(taskid:number):cfg.task.task2
---@field all table<any,cfg.task.task2>
---@field NullableRefTaskid cfg.task.taskextraexp
---@field NullableRefNexttask cfg.task.task

local this = cfg.task.task2

local mk = cfg._mk.i18n_table(this, { { 'all', 'get', 1 }, }, nil, { 
    { 'NullableRefTaskid', 0, cfg.task.taskextraexp, 'get', 1 }, 
    { 'NullableRefNexttask', 0, cfg.task.task, 'get', 3 }, }, 
    { name = 2 },
    'taskid', -- int, 任务完成条件类型（id的范围为1-100）
    'name', -- list<text>
    'nexttask', -- int
    'completecondition', -- task.completecondition
    'exp', -- int
    'testBool', -- bool
    'testString', -- str
    'testStruct', -- Position
    'testList', -- list<int>
    'testListStruct', -- list<Position>
    'testListInterface' -- list<ai.TriggerTick>
    )

local constvalue = Beans.ai.triggertick.constvalue
local position = Beans.position
local chat = Beans.task.completecondition.chat
local collectitem = Beans.task.completecondition.collectitem
local conditionand = Beans.task.completecondition.conditionand
local killmonster = Beans.task.completecondition.killmonster
local talknpc = Beans.task.completecondition.talknpc
local testnocolumn = Beans.task.completecondition.testnocolumn

local E = cfg._mk.E

local A = {}
A[1] = killmonster(1, 3)
A[2] = talknpc(1)
A[3] = collectitem(11, 1)
A[4] = position(0, 0, 0)
A[5] = testnocolumn

mk(1, {40}, 2, conditionand(conditionand(killmonster(1, 0), collectitem(0, 0)), talknpc(0)), 10001, true, "", position(1, 0, 0), {12}, {position(1, 2, 3)}, {constvalue(1234)})
mk(2, {41, 42}, 3, A[2], 2000, false, "", A[4], E, E, E)
mk(3, {43, 44}, 0, A[3], 3000, false, "", A[4], E, E, E)
mk(4, {45, 46}, 0, conditionand(A[1], A[3]), 4000, false, "", A[4], E, E, E)
mk(5, {47, 48}, 0, conditionand(conditionand(A[1], A[2]), A[3]), 5000, false, "", A[4], E, E, E)
mk(6, {49, 50}, 0, conditionand(chat("葵花宝典,123"), A[1]), 5000, false, "", A[4], E, E, E)
mk(7, {51, 52}, 0, A[5], 2000, false, "", A[4], E, E, E)
mk(8, {53, 54}, 0, A[5], 3000, false, "", A[4], E, E, E)

return this
